rotkehlchen/api/v1/schemas.py

```diff
@@ -2923,3 +2926,72 @@ def validate_schema(
                     message=f'address {data["associated_address"]} provided is not tracked by rotki for {data["evm_chain"]}',  # noqa: E501
                     field_name='associated_address',
                 )
+
+
+class BinanceSavingsSchema(BaseStakingQuerySchema):
+    location = LocationField(
+        required=True,
+        limit_to=[Location.BINANCE, Location.BINANCEUS],
+    )
+
+    def __init__(
+            self,
+            treat_eth2_as_eth: bool,
+    ) -> None:
+        super().__init__()
+        self.treat_eth2_as_eth = treat_eth2_as_eth
+
+    @post_load
+    def make_binance_savings_query(
```

Comment:
What in the hell are you doign here? Did you copy paste the code of the base class's post-load? Why?

Why is all this needed? You don't need all this logic. And even if you did then jsut call the base class. Don't just copy paste.